### ISSUE TITLE
v_generate_user_grant_revoke_ddl fails with "Invalid Operation: value too long"

### ISSUE THREAD

**Issue by sworisbreathing** [awslabs/amazon-redshift-utils#459]
I'm trying to remove a user from redshift using the `v_generate_user_grant_revoke_ddl` script. However when I try to run it like so:

```sql
select regexp_replace(ddl,grantor,'awsuser')
from admin.v_generate_user_grant_revoke_ddl
where grantor='foo'
and ddltype='grant'
and objtype <>'default acl'
order by objseq,grantseq;
```

it fails with:

```
ERROR:  value too long for type character varying(74)
```

If I use the redshift jdbc driver I get slightly more error info, but still not enough to know what to fix:

```
SQL Error [500310] [22001]: [Amazon](500310) Invalid operation: value too long for type character varying(74);
```

I've had a look through the view definition and can't find any reference to a `varchar(74)`, so I'm kind of at a loss here. It only happens for this one particular user ID - I've used it successfully in the past and even on my own userid it seems to run okay.

**Comment by sworisbreathing** [comment]
Okay, with a bit of effort, I was able to find a workaround, but I still don't understand what was causing the problem.

The view comprises what are essentially 3 main queries that are concatenated via `UNION ALL`. I was able to work around by splitting them up and running them individually, like so:

Query 1:
```sql
WITH generate_ddl AS (
	WITH objprivs AS ( 
		SELECT objowner, 
		schemaname, 
		objname, 
		objtype,
		CASE WHEN split_part(aclstring,'=',1)='' THEN 'PUBLIC' ELSE translate(trim(split_part(aclstring,'=',1)),'"','') END::text AS grantee,
		translate(trim(split_part(aclstring,'/',2)),'"','')::text AS grantor, 
		trim(split_part(split_part(aclstring,'=',2),'/',1))::text AS privilege, 
		CASE WHEN objtype = 'default acl' THEN objname 
		WHEN objtype = 'function' AND regexp_instr(schemaname,'[^a-z]') > 0 THEN objname
		WHEN objtype = 'function' THEN QUOTE_IDENT(schemaname)||'.'||objname 
		ELSE nvl(QUOTE_IDENT(schemaname)||'.'||QUOTE_IDENT(objname),QUOTE_IDENT(objname)) END::text as fullobjname,
		CASE WHEN split_part(aclstring,'=',1)='' THEN 'PUBLIC' 
		ELSE trim(split_part(aclstring,'=',1)) 
		END::text as splitgrantee,
		grantseq 
		FROM (
			-- TABLE AND VIEW privileges
			SELECT pg_get_userbyid(b.relowner)::text AS objowner, 
			trim(c.nspname)::text AS schemaname,  
			b.relname::text AS objname,
			CASE WHEN relkind='r' THEN 'table' ELSE 'view' END::text AS objtype, 
			TRIM(SPLIT_PART(array_to_string(b.relacl,','), ',', NS.n))::text AS aclstring, 
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(relacl,1))  AS n FROM pg_class) NS
			inner join pg_class B ON b.oid = ns.oid AND  NS.n <= array_upper(b.relacl,1)
			join pg_namespace c on b.relnamespace = c.oid
			where relkind in ('r','v')
			UNION ALL
			-- SCHEMA privileges
			SELECT pg_get_userbyid(b.nspowner)::text AS objowner,
			null::text AS schemaname,
			b.nspname::text AS objname,
			'schema'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.nspacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(nspacl,1)) AS n FROM pg_namespace) NS
			inner join pg_namespace B ON b.oid = ns.oid AND NS.n <= array_upper(b.nspacl,1)
			UNION ALL
			-- DATABASE privileges
			SELECT pg_get_userbyid(b.datdba)::text AS objowner,
			null::text AS schemaname,
			b.datname::text AS objname,
			'database'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.datacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(datacl,1)) AS n FROM pg_database) NS
			inner join pg_database B ON b.oid = ns.oid AND NS.n <= array_upper(b.datacl,1) 
			UNION ALL
			-- FUNCTION privileges 
			SELECT pg_get_userbyid(b.proowner)::text AS objowner,
			trim(c.nspname)::text AS schemaname, 
			textin(regprocedureout(b.oid::regprocedure))::text AS objname,
			'function'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.proacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq  
			FROM 
			(SELECT oid,generate_series(1,array_upper(proacl,1)) AS n FROM pg_proc) NS
			inner join pg_proc B ON b.oid = ns.oid and NS.n <= array_upper(b.proacl,1)
			join pg_namespace c on b.pronamespace=c.oid 
			UNION ALL
			-- LANGUAGE privileges
			SELECT null::text AS objowner,
			null::text AS schemaname,
			lanname::text AS objname,
			'language'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.lanacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq 
			FROM 
			(SELECT oid,generate_series(1,array_upper(lanacl,1)) AS n FROM pg_language) NS
			inner join pg_language B ON b.oid = ns.oid and NS.n <= array_upper(b.lanacl,1)
			UNION ALL
			-- DEFAULT ACL privileges
			SELECT pg_get_userbyid(b.defacluser)::text AS objowner,
			trim(c.nspname)::text AS schemaname,
			decode(b.defaclobjtype,'r','tables','f','functions')::text AS objname,
			'default acl'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.defaclacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq 
			FROM 
			(SELECT oid,generate_series(1,array_upper(defaclacl,1)) AS n FROM pg_default_acl) NS
			join pg_default_acl b ON b.oid = ns.oid and NS.n <= array_upper(b.defaclacl,1) 
			left join  pg_namespace c on b.defaclnamespace=c.oid
		) 
		where  (split_part(aclstring,'=',1) <> split_part(aclstring,'/',2) 
		and split_part(aclstring,'=',1) <> 'rdsdb'
		and NOT (split_part(aclstring,'=',1)='' AND split_part(aclstring,'/',2) = 'rdsdb')) 
	)
	-- Extract object GRANTS
	SELECT objowner, schemaname, objname, objtype, grantor, grantee, 'grant' AS ddltype, grantseq,
	decode(objtype,'database',0,'schema',1,'language',1,'table',2,'view',2,'function',2,'default acl',3) AS objseq,
	CASE WHEN (grantor <> current_user AND grantor <> 'rdsdb' AND objtype <> 'default acl') THEN 'SET SESSION AUTHORIZATION '||QUOTE_IDENT(grantor)||';' ELSE '' END::text||
	(CASE WHEN privilege = 'arwdRxt' OR privilege = 'a*r*w*d*R*x*t*' THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT ALL on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN privilege = 'a*r*w*d*R*x*t*' THEN ' with grant option;' ELSE ';' END::text) 
	when privilege = 'UC' OR privilege = 'U*C*' THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT ALL on '||objtype||' '||fullobjname||' to '||splitgrantee||
	(CASE WHEN privilege = 'U*C*' THEN ' with grant option;' ELSE ';' END::text) 
	when privilege = 'CT' OR privilege = 'U*C*' THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT ALL on '||objtype||' '||fullobjname||' to '||splitgrantee||
	(CASE WHEN privilege = 'C*T*' THEN ' with grant option;' ELSE ';' END::text)
	ELSE  
	(
	CASE WHEN charindex('a',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT INSERT on '||fullobjname||' to '||splitgrantee|| 
	(CASE WHEN charindex('a*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('r',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT SELECT on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('r*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('w',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT UPDATE on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('w*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('d',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT DELETE on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('d*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('R',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT RULE on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('R*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('x',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT REFERENCES on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('x*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('t',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT TRIGGER on '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('t*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('U',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT USAGE on '||objtype||' '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('U*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('C',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT CREATE on '||objtype||' '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('C*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('T',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT TEMP on '||objtype||' '||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('T*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text||
	CASE WHEN charindex('X',privilege) > 0 THEN (CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	ELSE '' END::text)||'GRANT EXECUTE on '||
	(CASE WHEN objtype = 'default acl' THEN '' ELSE objtype||' ' END::text)||fullobjname||' to '||splitgrantee||
	(CASE WHEN charindex('X*',privilege) > 0 THEN ' with grant option;' ELSE ';' END::text) ELSE '' END::text
	) END::text)|| 
	CASE WHEN (grantor <> current_user AND grantor <> 'rdsdb' AND objtype <> 'default acl') THEN 'RESET SESSION AUTHORIZATION;' ELSE '' END::text AS ddl
	FROM objprivs
)
SELECT regexp_replace(ddl,grantor,'awsuser')
FROM generate_ddl
where grantor='foo'
and ddltype='grant'
and objtype <>'default acl'
order by objseq,grantseq;
```

Query 2:
```sql
WITH generate_ddl AS (
	WITH objprivs AS ( 
		SELECT objowner, 
		schemaname, 
		objname, 
		objtype,
		CASE WHEN split_part(aclstring,'=',1)='' THEN 'PUBLIC' ELSE translate(trim(split_part(aclstring,'=',1)),'"','') END::text AS grantee,
		translate(trim(split_part(aclstring,'/',2)),'"','')::text AS grantor, 
		trim(split_part(split_part(aclstring,'=',2),'/',1))::text AS privilege, 
		CASE WHEN objtype = 'default acl' THEN objname 
		WHEN objtype = 'function' AND regexp_instr(schemaname,'[^a-z]') > 0 THEN objname
		WHEN objtype = 'function' THEN QUOTE_IDENT(schemaname)||'.'||objname 
		ELSE nvl(QUOTE_IDENT(schemaname)||'.'||QUOTE_IDENT(objname),QUOTE_IDENT(objname)) END::text as fullobjname,
		CASE WHEN split_part(aclstring,'=',1)='' THEN 'PUBLIC' 
		ELSE trim(split_part(aclstring,'=',1)) 
		END::text as splitgrantee,
		grantseq 
		FROM (
			-- TABLE AND VIEW privileges
			SELECT pg_get_userbyid(b.relowner)::text AS objowner, 
			trim(c.nspname)::text AS schemaname,  
			b.relname::text AS objname,
			CASE WHEN relkind='r' THEN 'table' ELSE 'view' END::text AS objtype, 
			TRIM(SPLIT_PART(array_to_string(b.relacl,','), ',', NS.n))::text AS aclstring, 
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(relacl,1))  AS n FROM pg_class) NS
			inner join pg_class B ON b.oid = ns.oid AND  NS.n <= array_upper(b.relacl,1)
			join pg_namespace c on b.relnamespace = c.oid
			where relkind in ('r','v')
			UNION ALL
			-- SCHEMA privileges
			SELECT pg_get_userbyid(b.nspowner)::text AS objowner,
			null::text AS schemaname,
			b.nspname::text AS objname,
			'schema'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.nspacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(nspacl,1)) AS n FROM pg_namespace) NS
			inner join pg_namespace B ON b.oid = ns.oid AND NS.n <= array_upper(b.nspacl,1)
			UNION ALL
			-- DATABASE privileges
			SELECT pg_get_userbyid(b.datdba)::text AS objowner,
			null::text AS schemaname,
			b.datname::text AS objname,
			'database'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.datacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq
			FROM 
			(SELECT oid,generate_series(1,array_upper(datacl,1)) AS n FROM pg_database) NS
			inner join pg_database B ON b.oid = ns.oid AND NS.n <= array_upper(b.datacl,1) 
			UNION ALL
			-- FUNCTION privileges 
			SELECT pg_get_userbyid(b.proowner)::text AS objowner,
			trim(c.nspname)::text AS schemaname, 
			textin(regprocedureout(b.oid::regprocedure))::text AS objname,
			'function'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.proacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq  
			FROM 
			(SELECT oid,generate_series(1,array_upper(proacl,1)) AS n FROM pg_proc) NS
			inner join pg_proc B ON b.oid = ns.oid and NS.n <= array_upper(b.proacl,1)
			join pg_namespace c on b.pronamespace=c.oid 
			UNION ALL
			-- LANGUAGE privileges
			SELECT null::text AS objowner,
			null::text AS schemaname,
			lanname::text AS objname,
			'language'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.lanacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq 
			FROM 
			(SELECT oid,generate_series(1,array_upper(lanacl,1)) AS n FROM pg_language) NS
			inner join pg_language B ON b.oid = ns.oid and NS.n <= array_upper(b.lanacl,1)
			UNION ALL
			-- DEFAULT ACL privileges
			SELECT pg_get_userbyid(b.defacluser)::text AS objowner,
			trim(c.nspname)::text AS schemaname,
			decode(b.defaclobjtype,'r','tables','f','functions')::text AS objname,
			'default acl'::text AS objtype,
			TRIM(SPLIT_PART(array_to_string(b.defaclacl,','), ',', NS.n))::text AS aclstring,
			NS.n as grantseq 
			FROM 
			(SELECT oid,generate_series(1,array_upper(defaclacl,1)) AS n FROM pg_default_acl) NS
			join pg_default_acl b ON b.oid = ns.oid and NS.n <= array_upper(b.defaclacl,1) 
			left join  pg_namespace c on b.defaclnamespace=c.oid
		) 
		where  (split_part(aclstring,'=',1) <> split_part(aclstring,'/',2) 
		and split_part(aclstring,'=',1) <> 'rdsdb'
		and NOT (split_part(aclstring,'=',1)='' AND split_part(aclstring,'/',2) = 'rdsdb')) 
	)
	-- Extract object REVOKES
	SELECT objowner, schemaname, objname, objtype, grantor, grantee, 'revoke'::text AS ddltype, grantseq,
	decode(objtype,'default acl',0,'function',1,'table',1,'view',1,'schema',2,'language',2,'database',3) AS objseq,
	CASE WHEN (grantor <> current_user AND grantor <> 'rdsdb' AND objtype <> 'default acl' AND grantor <> objowner) THEN 'SET SESSION AUTHORIZATION '||QUOTE_IDENT(grantor)||';' ELSE '' END::text||
	(CASE WHEN objtype = 'default acl' THEN 'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(grantor)||nvl(' in schema '||QUOTE_IDENT(schemaname)||' ',' ')
	||'REVOKE ALL on '||fullobjname||' FROM '||splitgrantee||';'
	ELSE 'REVOKE ALL on '||(CASE WHEN objtype = 'table' OR objtype = 'view' THEN '' ELSE objtype||' ' END::text)||fullobjname||' FROM '||splitgrantee||';' END::text)||
	CASE WHEN (grantor <> current_user AND grantor <> 'rdsdb' AND objtype <> 'default acl' AND grantor <> objowner) THEN 'RESET SESSION AUTHORIZATION;' ELSE '' END::text AS ddl
	FROM objprivs
	WHERE NOT (objtype = 'default acl' AND grantee = 'PUBLIC' and objname='functions')
)
SELECT regexp_replace(ddl,grantor,'awsuser')
FROM generate_ddl
where grantor='foo'
and ddltype='grant'
and objtype <>'default acl'
order by objseq,grantseq;
```

Query 3:
```sql
WITH generate_ddl AS (
    -- Eliminate empty default ACLs
	SELECT null::text AS objowner, null::text AS schemaname, decode(b.defaclobjtype,'r','tables','f','functions')::text AS objname,
			'default acl'::text AS objtype,  pg_get_userbyid(b.defacluser)::text AS grantor, null::text AS grantee, 'revoke'::text AS ddltype, 5 as grantseq, 5 AS objseq,
	  'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||' GRANT ALL on '||decode(b.defaclobjtype,'r','tables','f','functions')||' TO '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||
	CASE WHEN b.defaclobjtype = 'f' then ', PUBLIC;' ELSE ';' END::text AS ddl FROM pg_default_acl b where b.defaclnamespace=0
)
SELECT regexp_replace(ddl,grantor,'awsuser')
FROM generate_ddl
where grantor='foo'
and ddltype='grant'
and objtype <>'default acl'
order by objseq,grantseq;
```

Running them individually worked. I just don't understand why it fails when trying to run them together with the `UNION ALLs`.

**Comment by CrisFavero** [comment]
I too am running into this issue. 

if you run the individual component selects of the union and look at the metadata around the table definition  it is obvious that the culprit of reducing the ddl column size from 65k to 74 is the final select 

https://github.com/awslabs/amazon-redshift-utils/blob/master/src/AdminViews/v_generate_user_grant_revoke_ddl.sql#L204-L207
```SQL
SELECT null::text AS objowner, null::text AS schemaname, decode(b.defaclobjtype,'r','tables','f','functions')::text AS objname,
		'default acl'::text AS objtype,  pg_get_userbyid(b.defacluser)::text AS grantor, null::text AS grantee, 'revoke'::text AS ddltype, 5 as grantseq, 5 AS objseq,
  'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||' GRANT ALL on '||decode(b.defaclobjtype,'r','tables','f','functions')||' TO '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||
CASE WHEN b.defaclobjtype = 'f' then ', PUBLIC;' ELSE ';' END::text AS ddl FROM pg_default_acl b where b.defaclnamespace=0; 
```
null::text||
It would appear that Altering the concatenation line starting with `'ALTER DEFAULT PRIVILEGES for user ' || ...`  to include a `null::text||`

**Comment by CrisFavero** [comment]
I too am running into this issue. 

if you run the individual component selects of the union and look at the metadata around the table definition  it is obvious that the culprit of reducing the ddl column size from 65k to 74 is the final select 

https://github.com/awslabs/amazon-redshift-utils/blob/master/src/AdminViews/v_generate_user_grant_revoke_ddl.sql#L204-L207
```SQL
SELECT null::text AS objowner, null::text AS schemaname, decode(b.defaclobjtype,'r','tables','f','functions')::text AS objname,
		'default acl'::text AS objtype,  pg_get_userbyid(b.defacluser)::text AS grantor, null::text AS grantee, 'revoke'::text AS ddltype, 5 as grantseq, 5 AS objseq,
  'ALTER DEFAULT PRIVILEGES for user '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||' GRANT ALL on '||decode(b.defaclobjtype,'r','tables','f','functions')||' TO '||QUOTE_IDENT(pg_get_userbyid(b.defacluser))||
CASE WHEN b.defaclobjtype = 'f' then ', PUBLIC;' ELSE ';' END::text AS ddl FROM pg_default_acl b where b.defaclnamespace=0; 
```
It would appear that Altering the concatenation line starting with `'ALTER DEFAULT PRIVILEGES for user ' || ...`  to utilize strict casting to a varchar(65535) will 

simple test cases for validating character lengths of a string:
```SQL
select  'hi','hi'::text,CAST('hi' AS TEXT),CAST('hi'AS VARCHAR),CAST('hi' AS VARCHAR(65535)), 'hi'::varchar(65535);
```

**Comment by adedotua** [comment]
This should be fixed in #471 . You can test with the latest version and see if there are still issues.

**Comment by sworisbreathing** [comment]
@adedotua I vaguely remember having an issue with the latest version of the script but didn't capture the error message. Will let you know if I run across it again. If it's the same error I'll update this ticket otherwise will raise a new one for a different error

**Comment by saeedma8** [comment]
As mentioned above, the issue was fixed with #471. Please open a new issue if you face any other problem. Thank you.
